File: SelfConference/Categories/UIView+MBProgressHUD.m
//
//  UIView+MBProgressHUD.m
//  SelfConference
//
//  Created by Jeff Burt on 5/20/15.
//  Copyright (c) 2015 Self Conference. All rights reserved.
//

#import "UIView+MBProgressHUD.h"
#import <MBProgressHUD/MBProgressHUD.h>

@implementation UIView (MBProgressHUD)

- (void)SC_showProgressHUD {
    [self SC_showProgressHUDWithTitle:nil];
}

- (void)SC_hideProgressHUD {
    [MBProgressHUD hideHUDForView:self animated:YES];
}

- (void)SC_showProgressHUDWithTitle:(NSString *)title {
    MBProgressHUD *progressHUD = [[MBProgressHUD alloc] initWithView:self];
    progressHUD.detailsLabelText = title;
    [progressHUD show:YES];
    [self addSubview:progressHUD];
}

@end

File: SelfConference/Categories/UIView+MBProgressHud.swift
//
//  UIView+MBProgressHud.swift
//  SelfConference
//
//  Created by Flavius on 5/26/19.
//  Copyright © 2019 Self Conference. All rights reserved.
//

import Foundation
import UIKit

extension UIView {
    
    @objc(SC_showProgressHUDWithTitle:)
    func showProgressHud(withTitle title: String? = nil) {
        let progressHud = MBProgressHUD(view: self)
        progressHud.detailsLabel.text = title
        progressHud.show(animated: true)
        
        addSubview(progressHud)
    }
    
    @objc(SC_hideProgressHUD)
    func hideProgressHud() {
        MBProgressHUD.hide(for: self, animated: true)
    }
}
